gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java

```diff
@@ -24,4 +35,858 @@
+   *     Region Resource</a>
+   */
+  enum RegionField {
+    CREATION_TIMESTAMP("creationTimestamp"),
+    DESCRIPTION("description"),
+    ID("id"),
+    NAME("name"),
+    QUOTAS("quotas"),
+    SELF_LINK("selfLink"),
+    STATUS("status"),
+    ZONES("zones"),
+    DEPRECATED("deprecated");
+
+    private final String selector;
+
+    RegionField(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
+      return selector;
+    }
+
+    static String selector(RegionField... fields) {
+      Set<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
+      fieldStrings.add(SELF_LINK.selector());
+      for (RegionField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * Fields of a Compute Engine Zone resource.
+   *
+   * @see <a href="https://cloud.google.com/compute/docs/reference/latest/zones#resource">Zone
+   *     Resource</a>
+   */
+  enum ZoneField {
+    CREATION_TIMESTAMP("creationTimestamp"),
+    DESCRIPTION("description"),
+    ID("id"),
+    MAINTENANCE_WINDOWS("maintenanceWindows"),
+    NAME("name"),
+    REGION("region"),
+    SELF_LINK("selfLink"),
+    STATUS("status"),
+    DEPRECATED("deprecated");
+
+    private final String selector;
+
+    ZoneField(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
+      return selector;
+    }
+
+    static String selector(ZoneField... fields) {
+      Set<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
+      fieldStrings.add(SELF_LINK.selector());
+      for (ZoneField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * Fields of a Compute Engine License resource.
+   *
+   * @see <a href="https://cloud.google.com/compute/docs/reference/latest/licenses#resource">License
+   *     Resource</a>
+   */
+  enum LicenseField {
+    CHARGES_USE_FEE("chargesUseFee"),
+    NAME("name"),
+    SELF_LINK("selfLink");
+
+    private final String selector;
+
+    LicenseField(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
+      return selector;
+    }
+
+    static String selector(LicenseField... fields) {
+      Set<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 1);
+      fieldStrings.add(SELF_LINK.selector());
+      for (LicenseField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * Base class for list filters.
+   */
+  abstract class ListFilter implements Serializable {
+
+    private static final long serialVersionUID = -238638392811165127L;
+
+    private final String field;
+    private final ComparisonOperator operator;
+    private final Object value;
+
+    enum ComparisonOperator {
```

Comment:
Yes we decided to stick to v1, I plan to support multiple filters as soon as the Beta API becomes v2. This should be quite simple.